source/monitoring/cluster-monitoring.txt

```diff
@@ -0,0 +1,130 @@
+.. _c-cluster-monitoring:
+
+==================
+Cluster Monitoring
+==================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, server, topology
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecols
+
+Overview
+--------
+
+This guide shows you how to use the {+driver-short+} to monitor server
+discovery and monitoring (SDAM) events in a MongoDB instance, replica
+set, or sharded cluster. These events occur when there are any changes
+in the state of the MongoDB instance or cluster that you are connected
+to.
+
+You might use information about SDAM events in your application to
+understand cluster changes, assess cluster health, or perform capacity
+planning.
+
+.. _c-subscribe-sdam:
+
+Subscribe to Events
+-------------------
+
+You can access details about SDAM events by subscribing to them
+in your application. To subscribe to an event, define an Application
+Performance Monitoring (APM) callback function to handle each event
+type you want to subscribe to. Use a ``mongoc_apm_callbacks_t`` object
+to register the list of APM callbacks with a client using the
+``mongoc_client_set_apm_callbacks()`` function.
+
+This code monitors server opening events by performing the following
+actions:
+
+- Defines a ``server_opening()`` APM callback function
+- Creates a ``mongoc_apm_callbacks_t`` object to store callbacks
+- Calls the ``mongoc_apm_set_server_opening_cb()`` function, which
+  stores a pointer to the provided APM callback function in the
+  ``mongoc_apm_callbacks_t`` object
+- Calls the ``mongoc_client_set_apm_callbacks()`` function, which registers
+  the callback in the ``mongoc_apm_callbacks_t`` object with the client
+
+.. literalinclude:: /includes/monitoring/sdam.c
+   :language: c
+   :copyable:
+   :linenos:
+   :emphasize-lines: 10, 30-32
+
+When you perform a database operation, the driver establishes a new connection to
+the server and your subscriber records the server opening event. The code outputs
+messages that resemble the following:
+
+.. code-block:: none
+   :copyable: false
+
+   Server opening on ac-rmuag0v-shard-00-00.gh0qg50.mongodb.net:27017
+   Server opening on ac-rmuag0v-shard-00-01.gh0qg50.mongodb.net:27017
+   Server opening on ac-rmuag0v-shard-00-02.gh0qg50.mongodb.net:27017
```

Comment:
These examples seem a bit noisy / unnecessarily specific ("ac-rmuag0v"? "gh0qg50"?). Can they be simplified?